packages/data/addon/adapters/facts/bard.ts

```diff
@@ -180,7 +181,9 @@ export default class BardFactsAdapter extends EmberObject implements NaviFactAda
       if (!moment.utc(start).isValid()) {
         throw new FactAdapterError(`Since operator only supports datetimes, '${start}' is invalid`);
       }
-      end = moment.utc('9999-12-31').startOf(filterGrain).toISOString();
+      const dataSourceConfig = getDataSource(request.dataSource) as FiliDataSource;
+      const sinceOperatorEnd = dataSourceConfig.options?.sinceOperatorEndPeriod;
```

Comment:
This will cause issues because of the grain attached to the period. Say we have a week grain for the query, then no period of days/months/years will match up, only something like `P4W` would. Maybe we can do something similar to the next line where we cut it off to the closest grain that aligns